core/win32/inject.c

```diff
@@ -246,13 +246,13 @@ inject_into_thread(HANDLE phandle, CONTEXT *cxt, HANDLE thandle, char *dynamo_pa
              * the dynamorio lib.
              */
             for (i = 0; i < MCXT_NUM_SIMD_SLOTS; i++) {
-                for (j = 0; j < IF_X64_ELSE(2, 4); j++) {
+                for (j = 0; j < XMM_REG_SIZE / sizeof(*bufptr); j++) {
                     *bufptr++ = CXT_XMM(cxt, i)->reg[j];
                 }
                 /* FIXME i#437: save ymm fields.  For now we assume we're
                  * not saving and we just skip the upper 128 bits.
                  */
-                bufptr += IF_X64_ELSE(2, 4);
+                bufptr += (YMM_REG_SIZE - XMM_REG_SIZE) / sizeof(*bufptr);
```

Comment:
These two expressions were identical before but now they're not?

Reply:
Correct. One is the xmm buffer increment, the other is what's left of the slot size. ymm_reg_size - xmm_reg_size == xmm_reg_size currently. But with AVX-512, it's zmm_reg_size - xmm_reg_size != xmm_reg_size.

I am also going to add an assertion ASSERT(MCXT_SIMD_SLOT_SIZE == YMM_REG_SIZE) at above.